card-component-app/UserCard.jsx

```diff
@@ -0,0 +1,28 @@
+const UserCard = (props) => {
+  const [email, setEmail] = React.useState('loading...')
+  const [firstname, setFirstName] = React.useState('loading...')
+  const [lastname, setLastName] = React.useState('loading...')
+  const [avatar, setAvatar] = React.useState('...loading')
+
+  React.useEffect(() => {
+    async function userInfo(userId) {
+      let response = await fetch(`https://reqres.in/api/users/${userId}`);
+      let data = await response.json()
+      let user = await data.data;
+
+      setEmail(user.email)
+      setFirstName(user.first_name)
+      setLastName(user.last_name)
+      setAvatar(user.avatar)
+    }
+    userInfo(props.userId)
+}, [props.userId])
```

Comment:
Nice work remembering to include this dependency in the dependency array - this is what makes the `re-render` possible for your search functionality!